test/test_cuda.py

```diff
@@ -78,13 +78,30 @@ def is_half(t):
 ]
 
 
+dtype_lookup = {
+    'Double': torch.float64,
+    'Float': torch.float32,
+    'Half': torch.float16,
+    'Byte': torch.uint8,
+    'Char': torch.int8,
+    'Short': torch.int16,
+    'Int': torch.int32,
+    'Long': torch.int64
+}
```

Comment:
I think it might be better right now to have something like
```
def cast_tensor(tensor, t):
    return t(tensor.size()).copy_(tensor)
...
('__lshift__', lambda t: torch.pow(2, cast_tensor(torch.arange(1, 5), t)),
        lambda t: [2], None, signed_types)
```
to avoid the dtype lookup from the class. 

In the future it'll probably be a good idea to change the tests so that each test is passed a (dtype, device, layout) tuple instead of a tensor constructor.